lib/cosmos/packets/packet.rb

```diff
@@ -629,7 +629,7 @@ def reset
     #   buffer of data and processors
     def clone
       packet = super()
-      if packet.instance_variable_get("@processors")
+      if packet.instance_variable_get("@processors".freeze)
         packet.instance_variable_set("@processors", packet.processors.clone)
         packet.processors.each do |processor_name, processor|
```

Comment:
Here's a simple summary about it: http://tmm1.net/ruby21-fstrings/
Here's a breakdown about performance: http://www.sitepoint.com/unraveling-string-key-performance-ruby-2-2/